### ISSUE TITLE
download_file progress bar

### ISSUE THREAD

**Issue by jsta** [DOI-USGS/sciencebasepy#11]
Is there any interest in adding a progress bar to `download_file`? 

![Screenshot from 2020-07-23 13-51-14](https://user-images.githubusercontent.com/7844578/88320608-d28c1c80-cceb-11ea-9007-72814c7cd23b.png)

**Comment by MattReimer** [comment]
Absolutely, but it's important to make sure it's optional somehow so if we're capturing output to a log file on a server we can turn it off or make it more useful than 1000 lines of `==============` 

My own sloppy attempt for our tools is here:
https://github.com/Riverscapes/sqlBRAT/blob/master/lib/util.py#L154-L226

In our case I keep a `NO_UI` environment variable and then change the behaviour to make it a little more log-friendly. This works for our automation pipelines but you could use something like a boolean parameter too.

**Comment by jsta** [comment]
Makes sense. Sounds like it should be off by default.
